### ISSUE TITLE
System.Windows.Markup.XamlParseException: Start up

### ISSUE THREAD

**Issue by muekoeff** [naseweis520/osu-Sync#3]
**[Posted by Kurokami](http://bit.ly/1Cnm8t2)**
## Exception 01

> System.Windows.Markup.XamlParseException: 'Provide value on 'System.Windows.Baml2006.TypeConverterMarkupExtension' threw an exception.' Line number '6' and line position '23'. ---> System.IO.FileFormatException: The image format is unrecognized. ---> System.Runtime.InteropServices.COMException: Exception from HRESULT: 0x88982F07
>    --- End of inner exception stack trace ---
>    at System.Windows.Media.PixelFormat.GetPixelFormat(SafeMILHandle bitmapSource)
>    at System.Windows.Media.Imaging.BitmapSource.UpdateCachedSettings()
>    at System.Windows.Media.Imaging.BitmapSource.set_WicSourceHandle(BitmapSourceSafeMILHandle value)
>    at System.Windows.Media.Imaging.BitmapFrameDecode.FinalizeCreation()
>    at System.Windows.Media.Imaging.BitmapSource.CompleteDelayedCreation()
>    at System.Windows.Media.Imaging.BitmapSource.get_WicSourceHandle()
>    at System.Windows.Media.Imaging.BitmapFrameDecode..ctor(Int32 frameNumber, BitmapCreateOptions createOptions, BitmapCacheOption cacheOption, BitmapFrameDecode frameDecode)
>    at System.Windows.Media.Imaging.BitmapDecoder.SetupFrames(BitmapDecoder decoder, ReadOnlyCollection`1 frames)
>    at System.Windows.Media.Imaging.BitmapDecoder.Initialize(BitmapDecoder decoder)
>    at System.Windows.Media.Imaging.BitmapDecoder..ctor(SafeMILHandle decoderHandle, BitmapDecoder decoder, Uri baseUri, Uri uri, Stream stream, BitmapCreateOptions createOptions, BitmapCacheOption cacheOption, Boolean insertInDecoderCache, Boolean isOriginalWritable, Stream uriStream, UnmanagedMemoryStream unmanagedMemoryStream, SafeFileHandle safeFilehandle)
>    at System.Windows.Media.Imaging.IconBitmapDecoder..ctor(SafeMILHandle decoderHandle, BitmapDecoder decoder, Uri baseUri, Uri uri, Stream stream, BitmapCreateOptions createOptions, BitmapCacheOption cacheOption, Boolean insertInDecoderCache, Boolean originalWritable, Stream uriStream, UnmanagedMemoryStream unmanagedMemoryStream, SafeFileHandle safeFilehandle)
>    at System.Windows.Media.Imaging.BitmapDecoder.CreateFromUriOrStream(Uri baseUri, Uri uri, Stream stream, BitmapCreateOptions createOptions, BitmapCacheOption cacheOption, RequestCachePolicy uriCachePolicy, Boolean insertInDecoderCache)
>    at System.Windows.Media.Imaging.BitmapFrame.CreateFromUriOrStream(Uri baseUri, Uri uri, Stream stream, BitmapCreateOptions createOptions, BitmapCacheOption cacheOption, RequestCachePolicy uriCachePolicy)
>    at System.Windows.Media.ImageSourceConverter.ConvertFrom(ITypeDescriptorContext context, CultureInfo culture, Object value)
>    at System.Windows.Baml2006.TypeConverterMarkupExtension.ProvideValue(IServiceProvider serviceProvider)
>    at MS.Internal.Xaml.Runtime.ClrObjectRuntime.CallProvideValue(MarkupExtension me, IServiceProvider serviceProvider)
>    --- End of inner exception stack trace ---
>    at System.Windows.Markup.XamlReader.RewrapException(Exception e, IXamlLineInfo lineInfo, Uri baseUri)
>    at System.Windows.Markup.WpfXamlLoader.Load(XamlReader xamlReader, IXamlObjectWriterFactory writerFactory, Boolean skipJournaledProperties, Object rootObject, XamlObjectWriterSettings settings, Uri baseUri)
>    at System.Windows.Markup.WpfXamlLoader.LoadBaml(XamlReader xamlReader, Boolean skipJournaledProperties, Object rootObject, XamlAccessLevel accessLevel, Uri baseUri)
>    at System.Windows.Markup.XamlReader.LoadBaml(Stream stream, ParserContext parserContext, Object parent, Boolean closeStream)
>    at System.Windows.Application.LoadComponent(Object component, Uri resourceLocator)
>    at osu_Sync.Window_Welcome.InitializeComponent()
>    at osu_Sync.Window_Welcome..ctor()
>    at osu_Sync.MainWindow.MainWindow_Loaded(Object sender, RoutedEventArgs e)
>    at System.Windows.RoutedEventHandlerInfo.InvokeHandler(Object target, RoutedEventArgs routedEventArgs)
>    at System.Windows.EventRoute.InvokeHandlersImpl(Object source, RoutedEventArgs args, Boolean reRaised)
>    at System.Windows.UIElement.RaiseEventImpl(DependencyObject sender, RoutedEventArgs args)
>    at System.Windows.UIElement.RaiseEvent(RoutedEventArgs e)
>    at System.Windows.BroadcastEventHelper.BroadcastEvent(DependencyObject root, RoutedEvent routedEvent)
>    at System.Windows.BroadcastEventHelper.BroadcastLoadedEvent(Object root)
>    at MS.Internal.LoadedOrUnloadedOperation.DoWork()
>    at System.Windows.Media.MediaContext.FireLoadedPendingCallbacks()
>    at System.Windows.Media.MediaContext.FireInvokeOnRenderCallbacks()
>    at System.Windows.Media.MediaContext.RenderMessageHandlerCore(Object resizedCompositionTarget)
>    at System.Windows.Media.MediaContext.RenderMessageHandler(Object resizedCompositionTarget)
>    at System.Windows.Media.MediaContext.Resize(ICompositionTarget resizedCompositionTarget)
>    at System.Windows.Interop.HwndTarget.OnResize()
>    at System.Windows.Interop.HwndTarget.HandleMessage(WindowMessage msg, IntPtr wparam, IntPtr lparam)
>    at System.Windows.Interop.HwndSource.HwndTargetFilterMessage(IntPtr hwnd, Int32 msg, IntPtr wParam, IntPtr lParam, Boolean& handled)
>    at MS.Win32.HwndWrapper.WndProc(IntPtr hwnd, Int32 msg, IntPtr wParam, IntPtr lParam, Boolean& handled)
>    at MS.Win32.HwndSubclass.DispatcherCallbackOperation(Object o)
>    at System.Windows.Threading.ExceptionWrapper.InternalRealCall(Delegate callback, Object args, Int32 numArgs)
>    at MS.Internal.Threading.ExceptionFilterHelper.TryCatchWhen(Object source, Delegate method, Object args, Int32 numArgs, Delegate catchHandler)
## Exception 02

> System.DllNotFoundException: Unable to load DLL 'dwmapi': The specified module could not be found. (Exception from HRESULT: 0x8007007E)
>    at MahApps.Metro.Native.UnsafeNativeMethods.DwmSetWindowAttribute(IntPtr hwnd, Int32 attr, Int32& attrValue, Int32 attrSize)
>    at MahApps.Metro.Behaviours.BorderlessWindowBehavior.WindowProc(IntPtr hwnd, Int32 msg, IntPtr wParam, IntPtr lParam, Boolean& handled)
>    at System.Windows.Interop.HwndSource.PublicHooksFilterMessage(IntPtr hwnd, Int32 msg, IntPtr wParam, IntPtr lParam, Boolean& handled)
>    at MS.Win32.HwndWrapper.WndProc(IntPtr hwnd, Int32 msg, IntPtr wParam, IntPtr lParam, Boolean& handled)
>    at MS.Win32.HwndSubclass.DispatcherCallbackOperation(Object o)
>    at System.Windows.Threading.ExceptionWrapper.InternalRealCall(Delegate callback, Object args, Int32 numArgs)
>    at MS.Internal.Threading.ExceptionFilterHelper.TryCatchWhen(Object source, Delegate method, Object args, Int32 numArgs, Delegate catchHandler)

**Comment by muekoeff** [comment]
**[Posted by Kurokami](https://osu.ppy.sh/forum/p/3803035)**
**Operating System.** Windows XP
**.NET Framework Version.** 3.5, 4.0

**Comment by MaxKruse** [comment]
Maybe Update the Framework to 4.5, since i dont know which Framework was used to compile

**Comment by muekoeff** [comment]
osu!Sync indeed is compiled for .NET 4.0.
Personally I think that the exception occures because of MahApps.Metro but I'll test both possible solutions tomorrow with XP Mode in Windows Virtual PC.
Thank you. :)

**Comment by muekoeff** [comment]
Exception successfully reconstructed.
- Exception occures after SplashScreen.
- Exception warning message ("Baka") closes ifself for no reason.
  ![baka message](https://cloud.githubusercontent.com/assets/3207399/6004393/be13a158-ab02-11e4-9b8f-8244734f7548.png)
  ![exception message](https://cloud.githubusercontent.com/assets/3207399/6004225/85be130c-ab01-11e4-9442-8bc9af8dd7cf.PNG)

@MaxKruse | It's not possible to install .NET 4.5 on Windows XP. :disappointed:

**Comment by Shadowthegod** [comment]
Dat message by the way. :3

**Comment by muekoeff** [comment]
Actually I thought that I already submitted this but well, it isn't here.
So, the problem was fixed by my latest commit (https://github.com/naseweis520/osu-Sync/commit/b091a1d95d09a7560211526ed41209a3fd66d38d) but another exception occured.
However, I've already tested this some hours ago and exited with a different exception - but I haven't got a screenshot of it.
![Finally the UI loaded](https://cloud.githubusercontent.com/assets/3207399/6087475/309a91ea-ae4c-11e4-8486-544a6ce31fc7.png)
![Exception message](https://cloud.githubusercontent.com/assets/3207399/6087476/309f39ca-ae4c-11e4-9c7e-1c2d17eb69ef.png)
